lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -639,6 +641,49 @@ def deregister_inbound_handler(handler)
   attr_accessor :receiver_thread # :nodoc:
   attr_accessor :dispatcher_thread # :nodoc:
   attr_accessor :waiters # :nodoc:
+  attr_accessor :tlv_log_file # :nodoc:
+  attr_accessor :tlv_logging_enabled # :nodoc:
+
+  def log_packet(packet, packet_type)
+    option = framework.datastore['SessionTlvLogging']
+    return if option.nil? || option.casecmp?('false')
+
+    if option.casecmp?('console') || option.casecmp?('true')
+      log_packet_to_console(packet, packet_type)
+    elsif option.start_with?('file:')
+      log_packet_to_file(packet, packet_type)
+    end
+  end
+
+  def log_packet_to_console(packet, packet_type)
+    if packet_type == :recv
+      print "\n%bluRECV%clr: #{packet.inspect}\n"
+    elsif packet_type == :send
+      print "\n%redSEND%clr: #{packet.inspect}\n"
+    end
+  end
+
+  def log_packet_to_file(packet, packet_type)
+    path = framework.datastore['SessionTlvLogging'].split('file:').last
+    pathname = ::Pathname.new(path)
+
+    begin
+      self.tlv_log_file ||= ::File.open(pathname, 'a+')
```

Comment:
Not a blocker: If the user changed log file location, this would still log to the old location. I believe that's what `handle_session_tlv_logging` is meant to handle potentially